### DISCUSSION TITLE
Structured validation errors

### DISCUSSION THREAD

**Discussion by micimize** [simolus3/drift#1221] [Ideas]
Instead of surfacing an error with just a string, the top-level `InvalidDataException` should expose the underlying field-specific errors. This would allow for form validation integrations, etc.

This could even just be the existing internal `VerificationMeta` mapping. 

https://github.com/simolus3/moor/blob/8f069c86f42a5b0d40719773a2da14968b6409da/moor/lib/src/runtime/data_verification.dart#L70-L74

**Comment by simolus3** [comment]
I've added the map from the error too the `InvalidDataException` class.
